process_report/processors/validate_pi_alias_processor.py

```diff
@@ -0,0 +1,23 @@
+from dataclasses import dataclass
+
+import pandas
+
+from process_report.invoices import invoice
+from process_report.processors import processor
+
+
+@dataclass
+class ValidatePIAliasProcessor(processor.Processor):
+    alias_map: dict
+
+    @staticmethod
+    def _validate_pi_aliases(dataframe: pandas.DataFrame, alias_dict: dict):
+        for pi, pi_aliases in alias_dict.items():
+            dataframe.loc[
+                dataframe[invoice.PI_FIELD].isin(pi_aliases), invoice.PI_FIELD
+            ] = pi
+
+        return dataframe
+
+    def _process(self):
+        self.data = self._validate_pi_aliases(self.data, self.alias_map)
```

Comment:
1. Why is `_validate_pi_aliases` a static method?
2. Because we directly modify the pandas dataframe, there's no reason to return or assign things. 

What if we wrote it like this:
```python
class ValidatePIAliasProcessor(processor.Processor):
    alias_map: dict

    def _validate_pi_aliases(self):
        for pi, pi_aliases in alias_map.items():
            self.data.loc[
                self.data[invoice.PI_FIELD].isin(pi_aliases), invoice.PI_FIELD
            ] = pi

    def _process(self):
        self._validate_pi_aliases()
```

Not that it matters much, but just curious as I have noticed this pattern in other places too.

Reply:
I have changed this function, and another one in the `add_institute_processor.py` to use `self.data`, instead of having `data` passed in as an argument. This does lead to some inconsistency with other sections of code, specifically the invoice code. 

@knikolla @naved001  I will make a larger, sweeping commit to make this code design consistent if you think it's worth it.

Reply:
@QuanMPhm eh, it's fine. let's deal with that later and we can focus on the important refactoring for now.

Reply:
++, this refactoring is happening to enable an effective implementation of the prepaid groups work, everything more stylistic can wait until that is done.